### PR TITLE
Feature/google cloud vision detect operators

### PR DESCRIPTION
**Pull request by antonimaciej**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/hooks/gcp_vision_hook.py [hunk 1]
```diff
@@ -435,14 +440,97 @@ def annotate_image(self, request, retry=None, timeout=None):
...
 
         response = client.annotate_image(request=request, retry=retry, timeout=timeout)
 
-        self.log.info('Image annotated')
+        self.log.info("Image annotated")
 
         return MessageToDict(response)
 
+    @GoogleCloudBaseHook.catch_http_exception
+    def text_detection(self, image, max_results=None, retry=None, timeout=None, additional_properties={}):
```
Comment (mik-laj) [inline airflow/contrib/hooks/gcp_vision_hook.py]: https://docs.python-guide.org/writing/gotchas/ Dict is mutable

File: airflow/contrib/operators/gcp_vision_operator.py [hunk 2]
```diff
@@ -663,11 +665,529 @@ def __init__(
...
+    :type image: dict or google.cloud.vision_v1.types.Image
+    :param max_results: Number of results to return.
+    :type max_results: int
+    :param retry: (Optional) A retry object used to retry requests. If `None` is
+        specified, requests will not be retried.
+    :type retry: google.api_core.retry.Retry
+    :param timeout: Number of seconds before timing out.
+    :type timeout: float
+    :param additional_properties: Additional properties to be set on the AnnotateImageRequest. See more:
+        https://googleapis.github.io/google-cloud-python/latest/vision/gapic/v1/types.html#google.cloud.vision_v1.types.AnnotateImageRequest
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_vision_operator.py]:
```suggestion
        :class:`google.cloud.vision_v1.types.AnnotateImageRequest`
```

File: airflow/contrib/hooks/gcp_vision_hook.py (hunk 1 above)
Comment (antonimaciej) [reply airflow/contrib/hooks/gcp_vision_hook.py]: Thanks, didn't know it works that way.
Changed it

File: airflow/contrib/operators/gcp_vision_operator.py (hunk 2 above)
Comment (antonimaciej) [reply airflow/contrib/operators/gcp_vision_operator.py]: Done

**Comment by antonimaciej** [comment]
external PR already open

**Pull request closed without merging**